### PR TITLE
Refactor cert generation into a service

### LINKED ISSUE #537: Refactor cert generation into its own service
/kind feature

**Describe the solution you'd like**
Following this discussion here https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/502/files#r252461480
It'll be cleaner to implement cert generation, in cluster actuator, in its own service following the patterns used for ec2 and elb services

cc @chuckha 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by ashish-amarnath**
**What this PR does / why we need it**:
Fixes a bug in user data generation for machines joining the controlplane.
Makes cert generation as an implementation of the service interface.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #537 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/aws/actuators/cluster/actuator.go [hunk 1]
```diff
@@ -62,62 +62,23 @@ func (a *Actuator) Reconcile(cluster *clusterv1.Cluster) error {
 
 	ec2svc := ec2.NewService(scope)
 	elbsvc := elb.NewService(scope)
+	certSvc := certificates.NewService(scope)
 
 	// Store some config parameters in the status.
```
Comment (chuckha) [inline pkg/cloud/aws/actuators/cluster/actuator.go]: does this comment still make sense?

File: pkg/cloud/aws/services/certificates/service.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+limitations under the License.
+*/
+
+package certificates
+
+import (
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/aws/actuators"
+)
+
+// Service holds a collection of interfaces.
```
Comment (chuckha) [inline pkg/cloud/aws/services/certificates/service.go]: I see this comment copy and pasted around and I don't think it's actually accurate anymore. Please replace this whole section of comment describing what the certificate service provides

File: pkg/cloud/aws/actuators/cluster/actuator.go (hunk 1 above)
Comment (ashish-amarnath) [reply pkg/cloud/aws/actuators/cluster/actuator.go]: Updated this to describe what the reconcileCertificates() does.
Other Reconcile* are kinda self-explanatory.

**Comment by chuckha** [comment]
/approve
leaving the lgtm for @detiber

File: pkg/cloud/aws/services/certificates/service.go (hunk 2 above)
Comment (ashish-amarnath) [reply pkg/cloud/aws/services/certificates/service.go]: Updated to
```
// Service groups certificate related operations together and allows
// certificate updates to be applied to the actuator scope.
```

**Pull request merged**